lib/danger_plugin.rb

```diff
@@ -11,34 +10,45 @@ module Danger
   # @tags ruby, rubocop, linter
   #
   class DangerRubocop < Plugin
-
-    # Runs Ruby files through Rubocop. Generates a `markdown` list of warnings. 
+    # Runs Ruby files through Rubocop. Generates a `markdown` list of warnings.
     #
     # @param   [String] files
-    #          A globbed string which should return the files that you want to 
-    #          run through, defaults to nil. If nil, modified and added files 
+    #          A globbed string which should return the files that you want to
+    #          run through, defaults to nil. If nil, modified and added files
     #          from the diff will be used.
     # @return  [void]
     #
-    def run(files=nil)
+    def run(files = nil)
       files_to_lint = files ? Dir.glob(files) : (modified_files + added_files)
       files_to_lint.select! { |f| f.end_with? 'rb' }
-      rubocop_results = files_to_lint.map { |f| JSON.parse(`bundle exec rubocop -f json #{f}`)['files'] }.flatten
-      offending_files = rubocop_results.select { |f| f['offenses'].count > 0 }
 
+      offending_files = rubocop(files_to_lint)
       return if offending_files.empty?
 
-      message = '### Rubocop violations'
-      message << 'File | Line | Reason |\n'
-      message << '| --- | ----- | ----- |\n'  
+      markdown offenses_message(offending_files)
+    end
 
-      offending_files.each do |f|
-        f['offenses'].each do |o|
-          message << "#{f['path']} | #{o['location']['line']} | #{o['message']} \n"
-        end
+    private
+
+    def rubocop(files_to_lint)
+      rubocop_results = files_to_lint.flat_map do |f|
+        JSON.parse(`bundle exec rubocop -f json #{f}`)['files']
       end
+      rubocop_results.select { |f| f['offenses'].count > 0 }
+    end
 
-      markdown message
+    def offenses_message(offending_files)
+      require 'terminal-table'
```

Comment:
output

---

### Rubocop violations

+--------------+------+----------------+
| File         | Line | Reason         |
+--------------+------+----------------+
| ruby_file.rb | 13   | Don't do that! |
+--------------+------+----------------+